Elevel/Application/Features/QuestionCommands/DeleteQuestionCommand.cs

```diff
@@ -0,0 +1,50 @@
+﻿using AutoMapper;
+using Elevel.Application.Interfaces;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elevel.Application.Features.QuestionCommands
+{
+    public class DeleteQuestionCommand
+    {
+        public class Request : IRequest<Response>
+        {
+            public Guid? Id { get; set; }
+        }
+
+        public class Handler : IRequestHandler<Request, Response>
+        {
+            private readonly IApplicationDbContext _context;
+            private readonly IMapper _mapper;
+            public Handler(IApplicationDbContext context, IMapper mapper)
+            {
+                _context = context;
+                _mapper = mapper;
+            }
+            public async Task<Response> Handle(Request request, CancellationToken cancelationtoken)
+            {
+                var question = await _context.Questions.FirstOrDefaultAsync(a => a.Id == request.Id, cancelationtoken);
+                if (question is null)
+                {
+                    return null;
+                }
+                _context.Questions.Remove(question);
```

Comment:
you removes this entity and in next row try to modify it. but after call saveChanges it will be removed from DB. it's different approaches, and if you've already had Deleted property, so you should use soft delete option.

Reply:
It has been chosen to do soft delete option by removing Line 34 and keeping only "question.Deleted = true"